framework/validators/FileValidator.php

```diff
@@ -327,19 +313,16 @@ public function isEmpty($value, $trim = false)
      */
     private function sizeToBytes($sizeStr)
     {
-        switch (substr($sizeStr, -1)) {
-            case 'M':
-            case 'm':
-                return (int) $sizeStr * 1048576;
-            case 'K':
-            case 'k':
-                return (int) $sizeStr * 1024;
-            case 'G':
-            case 'g':
-                return (int) $sizeStr * 1073741824;
-            default:
-                return (int) $sizeStr;
-        }
+        static $sizes = [
+            'K' => 1024,
+            'M' => 1048576,
+            'G' => 1073741824,
+        ];
+        return (int) $sizeStr * ArrayHelper::getValue(
```

Comment:
switch statements can look full blown, but are extremely helpful in fast understanding of dimensions.